SYCL/ESIMD/spec_const/spec_const_bool.cpp

```diff
@@ -0,0 +1,21 @@
+//==--------------- spec_const_bool.cpp  - DPC++ ESIMD on-device test -----===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows
+// REQUIRES: linux && gpu
+// RUN: %clangxx-esimd -fsycl -I%S/.. %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// UNSUPPORTED: cuda
+
+#define DEF_VAL true
+#define REDEF_VAL false
+#define STORE 0
+
+typedef bool spec_const_t;
```

Comment:
In other test cases the same type is used for scalar and container. Is the difference expected. If so could you please add some comment here.